### ISSUE TITLE
add copyWith method to models

### ISSUE THREAD

**Issue by kidusdev** [medz/dart-orm#403]
Hy

I encounter this redundant code in various places of my application
```dart
final e = someMonthlyExpense;

 Expenses(
      id: e.amount == amount ? e.id : 0,
      amount: e.amount/30,
      period: e.period,
      reason: e.reason,
      startDate: e.startDate,y
      endDate: e.endDate,
      createdAt: e.createdAt,
    );
```

here in the above example i am grouping expenses to their lower expense period. its like if i have 1500$ house rent expense and if i have to cover that in a month how much should i earn daily and its obviously 50$ and the same for yearly expense. and the new value is treated as daily expense and the only difference is the amount. it would be great if we can do this instead;

```dart
final e = someMonthlyExpense;

final dailyExpense = e.copyWith(amount: e.amount/30);
// this ^ is better and concise code than this
/**
 Expenses(
      id: e.amount == amount ? e.id : 0,
      amount: e.amount/30,
      period: e.period,
      reason: e.reason,
      startDate: e.startDate,y
      endDate: e.endDate,
      createdAt: e.createdAt,
    );
*/

```

i tried
```
Expenses.fromJson({...e.toJson(), "amount": e.amount! * difference});
```
it doesn't work because the `toJson` method changes the `createdAt` to `String` then when i use the `fromJson` method is complains `type 'String' is not a subtype of type 'DateTime?'`. the DateTime Field in the models should parse a `String` and `DateTime` types

NB. i added the `copyWith` method to the models but it resets everytime i change the schema.prisma and generate a client

**Comment by medz** [comment]
I'll take some time to look at it. In fact, another problem has arisen. FromJson does not handle time strings correctly.

**Comment by kidusdev** [comment]
> another problem has arisen. FromJson does not handle time strings correctly.

Yeah, i encountered that many time i added an extra step to parse string to dates and assigning it to the model in my apps before mentioning it here.

**Comment by kidusdev** [comment]
> another problem has arisen. FromJson does not handle time strings correctly.

Yeah, i encountered that many time i added an extra step to parse string to dates and assigning it to the model in my apps before mentioning it here.

**Comment by medz** [comment]
Closing as superseded by the new Dart-native ORM merged in #485. This issue belongs to the retired Prisma-based client or its earlier roadmap. The new implementation uses Dart model declarations, independent drivers and single-engine Dart migrations, with no Prisma engine dependency.

The old implementation is no longer being developed. This closure does not claim that every requested feature or platform has been implemented. Please open a new issue against the new implementation with a current reproduction or focused proposal; see the repository README and capability documentation for the supported scope.

Thank you for the report and discussion.
